capture/capture.yaml

```diff
@@ -57,64 +55,72 @@ components:
           enum: [pcap, pcapng]
           default: pcap
 
-    Capture.BasicFilter:
+    Capture.Filter:
       description: >-
-        A container for different types of basic capture filters.
+        Container for capture filters
       type: object
       required: [choice]
       properties:
         choice:
           type: string
-          enum: [mac_address, custom]
-        mac_address:
-          $ref: '#/components/schemas/Capture.MacAddressFilter'
-        custom:
-          $ref: '#/components/schemas/Capture.CustomFilter'
-        and_operator:
-          type: boolean
-          default: true
-        not_operator:
-          type: boolean
-          default: false
+          enum: [ethernet, vlan, ipv4]
+        ethernet:
+          $ref: '#/components/schemas/Capture.Ethernet'
+        vlan:
+          $ref: '#/components/schemas/Capture.Vlan'
+        ipv4:
+          $ref: '#/components/schemas/Capture.Ipv4'
 
-    Capture.MacAddressFilter:
-      description: >-
-        A container for a mac address capture filter.
+    Capture.Custom:
       type: object
-      required: [mac, filter]
       properties:
-        mac:
-          description: >-
-            The type of mac address filters.
-            This can be either source or destination.
-          type: string
-          enum: [source, destination]
-        filter:
+        offset:
           description: >-
-            The value of the mac address.
+            The byte offset to filter on
+          type: integer
+        value:
           type: string
+          format: hex
         mask:
-          description: >-
-            The value of the mask to be applied to the mac address.
           type: string
-
-    Capture.CustomFilter:
-      description: >-
-        A container for a custom capture filter.
+          format: hex
+    
+    Capture.Field:
       type: object
-      required: [filter, offset]
       properties:
-        filter:
-          description: >-
-            The value to filter on.
+        value:
           type: string
+          format: hex
         mask:
-          description: >-
-            The mask to be applied to the filter.
           type: string
-        offset:
-          description: >-
-            The offset in the packet to filter at.
-          type: integer
+          format: hex
+
+    Capture.Ethernet:
```

Comment:
Do we want to add all these protocol headers in this file itself?  Will become a fairly long file as we add the other headers.
Also, any particular reason why you've only added Ethernet, VLAN and IPv4?

Reply:
The protocol headers can be moved to separate files once the proposal is accepted.  I've only added ethernet, vlan, ipv4 and custom to illustrate how the capture.filters creation is similar to flow.packet.  Adding additional protocols is a trivial exercise, and an extension like x-replace can be introduced so we can reuse the flow.packet headers and just replace the pattern ref with the capture field ref.